client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -0,0 +1,62 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteOption, getSiteAdminUrl } from 'state/sites/selectors';
+import { addQueryArgs } from 'lib/route';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const getIframeUrl = ( siteAdminUrl, postId, postType ) => {
+	if ( postId ) {
+		return `${ siteAdminUrl }post.php?post=${ postId }&action=edit&calypsoify=1`;
+	}
+	if ( 'post' === postType ) {
+		return `${ siteAdminUrl }post-new.php?calypsoify=1`;
+	}
+	return `${ siteAdminUrl }post-new.php?post_type=${ postType }&calypsoify=1`;
+};
+
+class CalypsoifyIframe extends Component {
+	render() {
+		const { iframeUrl } = this.props;
+
+		return (
+			<Fragment>
+				{ /* eslint-disable-next-line wpcalypso/jsx-classname-namespace */ }
+				<div className="main main-column calypsoify is-iframe" role="main">
+					{ /* eslint-disable-next-line jsx-a11y/iframe-has-title, wpcalypso/jsx-classname-namespace */ }
+					<iframe className={ 'is-iframe-loaded' } src={ iframeUrl } />
+				</div>
+			</Fragment>
+		);
+	}
+}
+
+export default connect( ( state, ownProps ) => {
+	const siteId = getSelectedSiteId( state );
+	const frameNonce = getSiteOption( state, siteId, 'frame_nonce' ) || '';
+	const siteAdminUrl = getSiteAdminUrl( state, siteId );
+
+	const iframeUrl = addQueryArgs(
+		{
+			'frame-nonce': frameNonce,
+		},
+		getIframeUrl( siteAdminUrl, ownProps.postId, ownProps.postType )
```

Comment:
We can calculate the iframe url with a bit of lodash and remove `getIframeUrl`

```js
const iframeUrl = addQueryArgs( pickBy(
	{
		'post': postId,
		'action': ( postId && 'edit' ), //if postId, edit.
		'post_type': ( postType !== 'post' && postType ), //if not post, use postType
		'calypsoify': 1,
		'frame-nonce': frameNonce,
	} ),
	getSiteAdminUrl( state, siteId, postId ? 'post.php' : 'post-new.php' )
);
```

has a bit of a cleverness factor, but we can always add some comments ;)

Reply:
Thanks for the suggestion! Your approach looks better to me, refactored in 0b836f55e535d2523235105beeb442022be9987f.